File: Java/civitas/src/civitas/Sorpresa.java
package civitas;

import java.util.ArrayList;
public abstract class Sorpresa{
    private String texto;
    
    Sorpresa(String _texto){
        texto = _texto;
    }
    
    public abstract void aplicarAJugador(int actual, ArrayList<Jugador> todos);
   
    
    public void informe(int actual, ArrayList<Jugador> todos){
        Diario.getInstance().ocurreEvento("Se aplica sorpresa a " + todos.get(actual).getNombre()+
                "\nLa informacion de la sorpresa es " + this.toString());
    }
    
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        if(actual>=0 && actual<todos.size())
            return true;
        else
            return false;
    }

    public String getTexto() {
        return texto;
    }
    
    
    
    @Override
    public String toString(){
        return "Sorpresa { " + " texto= " + texto + " ";
    }
    
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/sorpresa.rb
#encoding: utf­8

require_relative 'mazo_sorpresa'
require_relative 'tablero'
module Civitas
  class Sorpresa
    
    attr_accessor :texto
    
    def initialize(_texto)#Para enviar a la carcel
      @texto = _texto
    end
    
    def informe(actual,todos)
      Diario.instance.ocurre_evento("Se aplica sorpresa a " + todos.at(actual).nombre)
    end
    
    def jugador_correcto(actual,todos)
      if(actual >= 0 && actual < todos.size())
        return true
      else
        return false
      end
    end
    
    def salir_del_mazo()
      if(@tipo == Civitas::TipoSorpresa::SALIRCARCEL)
        @mazo.inhabilitar_carta_especial(@mazo.ultima_sorpresa)
      end
    end  
    

    def to_ss()
      frase = "Sorpresa texto:  #{@texto}"
      return frase
    end
  
  end
 
end

